controllers/servers/host_definer/watcher/watcher_helper.py

```diff
@@ -0,0 +1,469 @@
+import base64
+import datetime
+import os
+from kubernetes import client, config, dynamic
+from kubernetes.client import api_client
+from kubernetes.client.rest import ApiException
+
+from controllers.common.csi_logger import get_stdout_logger
+from controllers.servers.host_definer.common import settings
+from controllers.servers.host_definer.common.types import DefineHostRequest
+from controllers.servers.host_definer.storage_manager.host import StorageHostManager
+import controllers.servers.host_definer.watcher.exceptions as exceptions
+
+SECRET_IDS = {}
+NODES = {}
+logger = get_stdout_logger()
+
+
+class WatcherHelper:
+    def __init__(self):
+        self.storage_host_manager = StorageHostManager()
+        self._load_cluster_configuration()
+        self.dynamic_client = self._get_dynamic_client()
+        self.storage_api = client.StorageV1Api()
+        self.core_api = client.CoreV1Api()
+        self.custom_object_api = client.CustomObjectsApi()
+        self.csi_nodes_api = self._get_csi_nodes_api()
+        self.host_definitions_api = self._get_host_definitions_api()
+
+    def _get_dynamic_client(self):
+        return dynamic.DynamicClient(api_client.ApiClient(
+            configuration=self._load_cluster_configuration()))
+
+    def _load_cluster_configuration(self):
+        return config.load_incluster_config()
+
+    def _get_csi_nodes_api(self):
+        return self.dynamic_client.resources.get(
+            api_version=settings.STORAGE_API_VERSION,
+            kind=settings.CSINODE_KIND)
+
+    def _get_host_definitions_api(self):
+        return self.dynamic_client.resources.get(
+            api_version=settings.CSI_IBM_BLOCK_API_VERSION,
+            kind=settings.HOSTDEFINITION_KIND)
+
+    def verify_nodes_defined(self, host_request):
+        for node in NODES:
+            host_request.node_id = self.get_node_id_from_node_name(node)
+            self._verify_host_defined_and_has_host_definition(host_request)
+
+    def _verify_host_defined_and_has_host_definition(self, host_request):
```

Comment:
please consider layout more like this:
```
def _create_host_definition(self, host_request):
        if self._is_host_defined(host_request):
            logger.info('hostdefinition for node id {} is in Ready phase'.format(host_request.node_id))
            return
        response = self._define_host(host_request)
        host_definition = self._create_host_definition_if_not_exist(host_request)
        if response.error_message:
            self._set_host_definition_status(host_request, settings.PENDING_CREATION_PHASE)
            self._add_event_with_error_message(self, response, host_definition)
        self._set_host_definition_status(host_request, settings.READY_PHASE)
```
I took `_add_event_when_response_has_error_message` from `host_defenition_watcher` because logically we already have `host_definition` in from `_create_host_definition_if_not_exist` (its a version of `_verify_host_definition` without the `set_host_definition_status`)